jlburke42-signoff-Launching and Firing

jlburke42 opened the pull request:
Requesting signoff for Launching and Firing

jlburke42 committed:
Add files via upload

Uploaded files from old branch for new pull request, deleted old branch to reduce clutter

jlburke42 committed:
Delete Documentation/Signoffs/CapstoneDrawing.png

Updating to 12V motor

jlburke42 committed:
Delete Documentation/Signoffs/MOTORDATA.png

Updating to 12V motor

jlburke42 committed:
Delete Documentation/Signoffs/MOTORDATAGraph.png

Updating to 12V motor

jlburke42 committed:
Delete Documentation/Signoffs/Transmotec-Image-755.png

Updating to 12V motor

jlburke42 committed:
Add files via upload

Updating to 12V motor

jlburke42 committed:
Update LaunchingAndFiring.md

Update to 12V motor

jlburke42 committed:
Update LaunchingAndFiring.md

Fixing picture alignment

jlburke42 committed:
Update LaunchingAndFiring.md

Updated constraints to reflect subsystem-specific constraints

JTJones73 commented:
put assets in ./Documentation/images
"continue to power the DC motor" what do you mean by continuing to power? To control or to piggy back off your subsystem
"12-volt DC brushed motor" I would probably discuss this in the bill of materials/analysis
" a relay " Why a relay? Would that not be difficult to precisely stop the actuator? Why not a mosfet, motor controller, or esc
Are you certain that the relay would not damage the pi? Would it be wise to add a optocoupler for further isolation?
You must have references especially for datasheets and figures
Resize  the image of the motor it is quite large
Talk more about what your subsystem is supposed to do. It is a but confusing, as more of the document is about a DC motor rather than how the materials will help you achieve your goal.

JesseTNRoberts commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -0,0 +1,63 @@
+| 3. | Fit within a cubic foot of space | The Launching and Firing subsystem shall fit within a cubic foot of space (1 ft by 1 ft by 1 ft) along with the other subsystems that will be placed within that space | DEVCOM competition rulebook |
+| 4. | Motor Torque | The Launching and Firing Mechanism shall operate the firing mechanism on the launcher apparatus using a moto of at least 28 mN-m | Mechanical Engineering Team Specification |
+| 5. |  Signal-to-Fire delay | The Launching and Firing subsystem shall fire a round within 80 ms after receiving the signal from the aiming subsystem. This is the amount of time left after accounting for all other subsystems to complete their processes during the worst case scenario. | Team Specification |
+| 6. | Preloading after firing | The Launching and Firing subsystem shall reset the preload condition after each round is fired. | Team Specification | 
```
How fast does this need to occur?

jlburke42 replied:
Ahh good point, thank you !

JesseTNRoberts reviewed (commented):
Since were moving away from the motor here, I'll wait to review with the new motor and controller

jlburke42 committed:
Delete Documentation/Signoffs/CapstoneDrawing.png

Moving images to images folder

jlburke42 committed:
Delete Documentation/Signoffs/MOTORDATA.png

Moving images to images folder

jlburke42 committed:
Delete Documentation/Signoffs/MOTORDATAGraph.png

Moving images to images folder

jlburke42 committed:
Delete Documentation/Signoffs/RelayCharachteristics.png

Moving images to images folder

jlburke42 committed:
Delete Documentation/Signoffs/Transmotec-Image-755.png

Moving images to images folder

jlburke42 committed:
Add files via upload

Moving images to images folder

jlburke42 committed:
Update LaunchingAndFiring.md

Updating image branches

jlburke42 committed:
Delete Documentation/Images/CapstoneDrawing.png

Updating Images

jlburke42 committed:
Delete Documentation/Images/MOTORDATA.png

Updating Images

jlburke42 committed:
Delete Documentation/Images/MOTORDATAGraph.png

Updating Images

jlburke42 committed:
Delete Documentation/Images/Transmotec-Image-755.png

Updating Images

jlburke42 committed:
Add files via upload

Updating Images

jlburke42 committed:
Update LaunchingAndFiring.md

Update image paths

jlburke42 committed:
Update LaunchingAndFiring.md

Updated with DigiKey motor

jlburke42 committed:
Update LaunchingAndFiring.md

Added better terminology to the functionality statement

JesseTNRoberts reviewed (commented):
One concern is that the raspberry pi needs to be represented once and every design that touches it updates that representation. This is to ensure that different designs don't double use pins.

This design is good. My main question is, how will you know when the firing mechanism has reset? The motor doesn't give you feedback. So, this is an open loop system (those never work very well).

jlburke42 committed:
Delete Documentation/Images/CapstoneDrawing.png

Updating drawing

jlburke42 committed:
Add files via upload

Updating drawing

jlburke42 committed:
Delete Documentation/Images/CapstoneDrawing.png

Updating drawing

jlburke42 committed:
Add files via upload

Updating drawing

jlburke42 committed:
Delete Documentation/Images/CapstoneDrawing.png

Updating drawing

jlburke42 committed:
Add files via upload

Updating drawing

jlburke42 committed:
Update LaunchingAndFiring.md

Updating drawing

jlburke42 committed:
Update LaunchingAndFiring.md

Updated electrical schematic, added explanation of DC motor signal control.

JesseTNRoberts reviewed (commented):
Should the constraint be that the firing mechanism fire within a certain time or should it be that it is capable of firing within that time? What if everything gets done early and firing now would miss?

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Update LaunchingAndFiring.md

Added proof for DC motor reliability for the timing of the round below constraint 5

jlburke42 committed:
Add files via upload

Added picture from PMDC motor study

jlburke42 committed:
Update LaunchingAndFiring.md

added clarity of PMDC in functionality

jlburke42 committed:
Update LaunchingAndFiring.md

Reference and links for parts used

JesseTNRoberts commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -0,0 +1,78 @@
+## Analysis 
+
+**Constraint Solution 1:**
+The Launching and Firing subsystem shall use 16 AWG wire for the subsystems electrical connections. This wire has an ampacity rating of 10 amps over a distance of 12 inches. This wire size was selected by considering the largest ampacity draw that exists in the subsystem. For this the team decided to use the start/stall current (9.8 amps) of the 12V DC motor used in this subsystem to apply mechan ...
```
A wire is only permitted to be filled to 80% of it's ampacity.

JesseTNRoberts commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -0,0 +1,78 @@
+**Constraint Solution 2:**
+To allow the firing mechanism to operate without human interaction, a signal shall be generated from the Aiming subsystem when that subsystem has finished the aiming processes that will be used as a relay signal to operate the firing mechanism autonomously. The relay will disconnect the 12-volt supply from the DC motor when the signal is not active, or low, and will connect the 12-volt supply to  ...
+
+Below is the charachteristics sheet of the Adafruit Featherwing Relay[3]:
```
How do you know the relay can handle the current?

JesseTNRoberts reviewed (commented):
I have a few questions below.

jlburke42 committed:
Update LaunchingAndFiring.md

changed 16 AWG wire to 14 AWG, also clarified relay information

jlburke42 committed:
Update LaunchingAndFiring.md

Fixing some errors left from using the older numbers from the ME team

jlburke42 committed:
Update LaunchingAndFiring.md

Small fixes

micahr234 commented on Documentation/Signoffs/LaunchingAndFiring.md:
How are you going to reliably stop at the same point every time? Each time you fire you will be off slightly. How will you compensate for the buildup of error?

When talking about torque, which side of the  gearbox are you referring to... the input or the output? Who is buying the gearbox?

jlburke42 replied:
Great questions, firstly, after weeks in f deliberation last semester we can to the cam idea with the ME team, they ensured that this mechanism will provide us with a consistent amount of resistance at the end of each cycle once the motor is no longer powered to allow us to stop within our margin of error. We will tune the motors on/off frequency in testing so that it is the most effective and efficient. Luckily, there is little to no torque at the beginning and we will basically just be timing the preload cycle - which will remain consistent throughout the competition. For this paper, all torques are the input shaft torques given by the motor and multiplied through the 30:1 torque converter, which will be built by the ME team

micahr234 replied:
Just having a cam will not ensure that the motor stops in the same place every time. Each time you fire you will be off slightly. How will you prevent the **buildup of error**? I have done similar application in my professional career. You MUST have a way to determine that the motor made it back to the stating position.

So when you talk about 3.2 OZ-IN of torque needed that is a reference to gearbox input? Right? Make this more clear in your documentation.

micahr234 replied:
Not sure what you have changed to address the concerns raised above.

micahr234 commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -0,0 +1,84 @@
+The Launching and Firing subsystem shall use 16 AWG wire for the subsystems electrical connections. This wire has an ampacity rating of 10 amps over a distance of 12 inches. This wire size was selected by considering the largest ampacity draw that exists in the subsystem. For this the team decided to use the start/stall current (9.8 amps) of the 12V DC motor used in this subsystem to apply mechan ...
+
+**Constraint Solution 2:**
+To allow the firing mechanism to operate without human interaction, a signal shall be generated from the Aiming subsystem when that subsystem has finished the aiming processes that will be used as a relay signal to operate the firing mechanism autonomously. The relay[3] will disconnect the 12-volt supply from the DC motor when the signal is not active, or low, and will connect the 12-volt supply  ...
```
Look at the relay again. What DC current can it handle? Note that relays can handle higher ac current than dc current.

jlburke42 replied:
Hey Professor, I was able to add a proximity sensor to the design and told the ME team this was to be incorporated into the design based on the placement of where they needed it to be, but electrically and for our purposes, this document has been updated. Also, I made notes of when geared torque was in question, what side of the gearing I was on. For the relay, the image below seems to point me to a switching current of 10A DC, and rated for 10A up to 35VDC, if you have any insight on how it may differ please let me know! Thanks
![image](https://github.com/user-attachments/assets/876843f5-b6db-46f6-b3bc-9f9e128ab4b3)

jlburke42 committed:
Add files via upload

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Add files via upload

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Rename Electrical Schematic.png to Electrical_Schematic.png

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Delete Documentation/Images/Electrical_Schematic.png

jlburke42 committed:
Add files via upload

jlburke42 committed:
Delete Documentation/Images/Electrical_Schematic.png

jlburke42 committed:
Add files via upload

jlburke42 committed:
Delete Documentation/Images/Electrical_Schematic.png

jlburke42 committed:
Add files via upload

jlburke42 committed:
Update LaunchingAndFiring.md

micahr234 commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -17,22 +17,26 @@ This subsystem will use a 12-volt DC brushed motor along with a relay that will
 
 **Constraint Solution 2:**
-To allow the firing mechanism to operate without human interaction, a signal shall be generated from the Aiming subsystem when that subsystem has finished the aiming processes that will be used as a relay signal to operate the firing mechanism autonomously. The relay[3] will disconnect the 12-volt supply from the DC motor when the signal is not active, or low, and will connect the 12-volt supply  ...
+To allow the firing mechanism to operate without human interaction, a signal shall be generated from the Aiming subsystem when that subsystem has finished the aiming processes that will be used as a relay signal to operate the firing mechanism autonomously. The relay[3] will disconnect the 12-volt supply from the DC motor when the signal is not active, or low, and will connect the 12-volt supply  ...
```
Will the raspberry pi take a 12v input from the prox? This prox sensor is a photo prox not an inductive prox. Will it be able to "see" a metal bolt?

micahr234 commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -1,8 +1,8 @@
 
 ## Explanation of DC Motor Signal Control
-This subsystem will use a 12-volt DC brushed motor along with a relay that will use a signal from the aiming subsystem as an input to allow the DC motor to operate. This signal will be sent for the amount of milliseconds that it will take the firing apparatus to fire a round, and return to the preloaded condition. This time will be found once the Mechancial Engineering team has finished construct ...
+This subsystem will use a 12-volt DC brushed motor along with a relay that will use a signal from the aiming subsystem as an input to allow the DC motor to operate. This signal will be sent for the amount of milliseconds that it will take the firing apparatus to fire a round, and return to the preloaded condition. This time will be found once the Mechancial Engineering team has finished construct ...
```
This needs to be rewritten to use the prox. It still reads as if it was time based.

micahr234 commented on Documentation/Signoffs/LaunchingAndFiring.md:
```diff
@@ -0,0 +1,93 @@
+The Launching and Firing subsystem shall use 16 AWG wire for the subsystems electrical connections. This wire has an ampacity rating of 10 amps over a distance of 12 inches. This wire size was selected by considering the largest ampacity draw that exists in the subsystem. For this the team decided to use the start/stall current (9.8 amps) of the 12V DC motor used in this subsystem to apply mechan ...
+
+**Constraint Solution 2:**
+To allow the firing mechanism to operate without human interaction, a signal shall be generated from the Aiming subsystem when that subsystem has finished the aiming processes that will be used as a relay signal to operate the firing mechanism autonomously. The relay[3] will disconnect the 12-volt supply from the DC motor when the signal is not active, or low, and will connect the 12-volt supply  ...
```
Is the Adafruit Power Relay 3191 using the G5LE-G or G5LE-14? Will the G5LE-14 carry 10A at 12VDC? Look carefully at the datasheets online.

jlburke42 committed:
Update LaunchingAndFiring.md

jlburke42 committed:
Delete Documentation/Images/CapstoneDrawing2.png

jlburke42 committed:
Delete Documentation/Images/Electrical_Schematic_new.png

jlburke42 committed:
Delete Documentation/Images/Prox_sensor_body.png

jlburke42 committed:
Delete Documentation/Images/RelayCharachteristics.png

jlburke42 committed:
Add files via upload
